packages/core/src/payment/strategies/braintree/braintree-paypal-payment-strategy.ts

```diff
@@ -223,6 +238,71 @@ export default class BraintreePaypalPaymentStrategy implements PaymentStrategy {
         };
     }
 
+    private async _loadPaypalCheckoutInstance(
+        braintreeOptions?: BraintreePaymentInitializeOptions,
+    ) {
+        const { clientToken, initializationData } = this._paymentMethod || {};
+
+        if (!clientToken || !initializationData) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        if (!initializationData.enablePaypalPaywallBanner) {
+            return Promise.resolve(this._store.getState());
+        }
+
+        try {
+            this._braintreePaymentProcessor.initialize(
+                clientToken,
+                initializationData,
+                braintreeOptions,
+            );
+
+            const currencyCode = this._store.getState().cart.getCartOrThrow().currency.code;
+
+            const paypalCheckoutConfig: Partial<BraintreePaypalSdkCreatorConfig> = {
+                currency: currencyCode,
+                intent: initializationData?.intent,
+                isCreditEnabled: initializationData?.isCreditEnabled,
+            };
+
+            await this._braintreePaymentProcessor.preloadPaypalCheckout(
+                paypalCheckoutConfig,
+                () => {
+                    this._renderPayPalMessages(braintreeOptions?.bannerContainerId);
+                },
+                this._handleError,
+            );
+        } catch (error) {
+            this._handleError(error);
+        }
+
+        return Promise.resolve(this._store.getState());
+    }
+
+    private _renderPayPalMessages(containerId = '') {
+        const isMessageContainerAvailable =
+            containerId && Boolean(document.getElementById(containerId));
+
+        if (this._braintreeHostWindow.paypal && isMessageContainerAvailable) {
+            const state = this._store.getState();
+            const cart = state.cart.getCartOrThrow();
+
+            this._braintreeHostWindow.paypal
+                .Messages({
+                    amount: cart.cartAmount,
```

Comment:
Think you can use subtotal from checkout object.

Reply:
Done